app/controllers/HomeController.scala

```diff
@@ -45,4 +45,23 @@ class HomeController @Inject()(val controllerComponents: ControllerComponents,pe
     }
   }
 
+  def createPermissions(userName: String): Action[AnyContent] = authAction { implicit request: UserRequest[AnyContent] =>
+    val requestedFiles = extractRequestedFiles(request)
+    try {
+      val (created, notCreated) = perms.createPermissions(request.token, userName, requestedFiles)
+      if (notCreated.nonEmpty) {
+        NotFound(notCreated.mkString("\n"))
+      } else {
+        Ok(created.mkString("\n"))
+      }
+    } catch {
+      case _: AuthorizationDeniedException => Forbidden(requestedFiles.mkString("\n"))
+      case e: Throwable => throw e
```

Comment:
would you better answer with an http status 500 instead of re-throw the error ? Or you have a global handler somewhere that will do it ?

Reply:
We can format a cleaner message indeed